This patch reverts the changes for cargo 0.49 and allows building with the
rust-cargo 0.43 package that is in Debian, it should be removed when cargo
0.49 is packaged.

Index: debcargo/src/crates.rs
===================================================================
--- debcargo.orig/src/crates.rs
+++ debcargo/src/crates.rs
@@ -3,6 +3,7 @@ use cargo::{
     core::manifest::ManifestMetadata,
     core::registry::PackageRegistry,
     core::source::MaybePackage,
+    core::InternedString,
     core::{
         Dependency, EitherManifest, FeatureValue, Manifest, Package, PackageId, Registry, Source,
         SourceId, Summary, Target, TargetKind, Workspace,
@@ -10,7 +11,7 @@ use cargo::{
     ops,
     ops::PackageOpts,
     sources::registry::RegistrySource,
-    util::{interning::InternedString, toml::read_manifest, FileLock},
+    util::{toml::read_manifest, FileLock},
     Config,
 };
 use filetime::{set_file_times, FileTime};
@@ -115,7 +116,7 @@ impl CrateInfo {
                     all_features: true,
                     no_default_features: false,
                     jobs: None,
-                    targets: Vec::new(),
+                    target: None,
                     features: Vec::new(),
                 };

@@ -565,10 +566,9 @@ impl CrateInfo {

         // Ensure that Cargo.toml is in standard form, e.g. does not contain
         // path dependencies, so can be built standalone (see #4030).
-        let toml_path = path.join("Cargo.toml");
-        let ws = Workspace::new(&toml_path.canonicalize()?, &self.config)?;
-        let registry_toml = self.package().to_registry_toml(&ws)?;
+        let registry_toml = self.package().to_registry_toml(&self.config)?;
         let mut actual_toml = String::new();
+        let toml_path = path.join("Cargo.toml");
         fs::File::open(&toml_path)?.read_to_string(&mut actual_toml)?;

         if !actual_toml.contains("AUTOMATICALLY GENERATED BY CARGO") {
Index: debcargo/Cargo.toml
===================================================================
--- debcargo.orig/Cargo.toml
+++ debcargo/Cargo.toml
@@ -25,7 +25,7 @@ version = "0.12"
 version = "1.0"

 [dependencies.cargo]
-version = "0.49"
+version = "0.43"

 [dependencies.chrono]
 version = "0.4"
